apps/sms/services/test/unit/moz_mobile_message/messaging_service_test.js

```diff
@@ -0,0 +1,670 @@
+/*global bridge,
+         MocksHelper
+*/
+
+'use strict';
+
+require('/services/js/bridge_service_mixin.js');
+require('/services/test/unit/mock_bridge.js');
+require('/views/shared/test/unit/mock_messages.js');
+require('/views/shared/test/unit/mock_settings.js');
+require('/views/shared/test/unit/mock_smil.js');
+
+var MocksHelperForAttachment = new MocksHelper([
+  'bridge',
+  'Settings',
+  'SMIL'
+]).init();
+
+var MessagingService, successEvt, errorEvt;
+
+suite('Messaging service >', function() {
+
+  MocksHelperForAttachment.attachTestHelpers();
+
+  suiteSetup(function(done) {
+    window.self = {};
```

Comment:
is it really necessary ? `self === window` in a window context so if we have the right global objects it should work fine ?